src/snakemake/sourcecache.py

```
@@ -189,19 +293,97 @@ def is_local(self):
+            shutil.move(tmpdir, self.repo_clone)
+
+    @property
+    def repo(self) -> "git.Repo":
+        return self._repo
+
+
 class HostingProviderFile(SourceFile):
     """Marker for denoting github source files from releases."""
 
     valid_repo = re.compile("^.+/.+$")
+    _hosted_repos: Dict[str, HostedGitRepo] = {}
```

Ruff (0.14.10)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)